src/vdp_scanner.py

```diff
@@ -0,0 +1,296 @@
+        """Output the agency results to a CSV."""
+        file = path_join(self.output_directory, self.agency_csv)
+        with open(file, "w") as csv_out:
+            agency_output = csv.DictWriter(
+                csv_out, fieldnames=VdpScanner.agency_csv_header
+            )
+            agency_output.writeheader()
+            for agency, info in self.agency_results.items():
+                output_dict = {"Agency": agency, **info}
+                agency_output.writerow(output_dict)
+
+    def output_domain_csv(self) -> None:
+        """Output the agency results to a CSV."""
+        file = path_join(self.output_directory, self.domain_csv)
+        with open(file, "w") as csv_out:
+            domain_output = csv.DictWriter(
+                csv_out, fieldnames=VdpScanner.domain_csv_header
+            )
+            domain_output.writeheader()
+            for result in self.domain_results:
+                domain_output.writerow(result)
+
+    def output_all_csvs(self) -> None:
+        """Output all CSVs."""
+        self.output_agency_csv()
+        self.output_domain_csv()
+
+
+def get_version(version_file) -> str:
+    """Extract a version number from the given file path."""
+    with open(version_file) as vfile:
+        for line in vfile.read().splitlines():
+            if line.startswith("__version__"):
+                delim = '"' if '"' in line else "'"
+                return line.split(delim)[1]
+
+    raise RuntimeError("Unable to find version string.")
+
+
+def get_local_csv(file: str) -> List[Dict[str, str]]:
+    """Load domains from a local CSV file."""
+    with open(file) as csv_file:
+        csv_lines = [line.rstrip() for line in csv_file.readlines()]
+
+    return list(csv.DictReader(csv_lines))
+
+
+def get_remote_csv() -> List[Dict[str, str]]:
+    """Load domains from the CSV at the given URL."""
+    resp = requests.get(
+        "https://raw.githubusercontent.com/GSA/data/master/dotgov-domains/current-federal.csv"
+    )
+    if resp.status_code != 200:
+        return []
+    csv_lines = [str(line, resp.encoding) for line in resp.iter_lines()]
+
+    return list(csv.DictReader(csv_lines))
+
+
+def main():
+    """Scan hosts with the hash-http-content package and output results."""
+    __version__: str = get_version("version.txt")
+    args: Dict[str, Any] = docopt.docopt(__doc__, version=__version__)
+
+    log_level = logging.DEBUG if args["--debug"] else logging.INFO
+    logging.basicConfig(
+        format="%(asctime)-15s %(levelname)s %(message)s", level=log_level
+    )
+
+    # If we make a call to UrlHasher.hash_url() with verify=False, it will output
+    # a warning. Since this is a fallback mechanism, we can squelch these warnings.
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+    browser_opts = {
+        "args": [
+            "--no-sandbox",
+            "--disable-gpu",
+            "--disable-dev-shm-usage",
+            "--no-zygote",
```

Comment:
Consider adding a comment explaining why each browser option argument is being used here.